File: src/my_cpp_pkg/src/publisher_test.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/string.hpp"

using namespace std::chrono_literals;

class PublisherTest : public rclcpp::Node 
{
public:
    PublisherTest() : Node("publisher_test") 
    {
        publisher_ = this->create_publisher<example_interfaces::msg::String>("my_topic" , 10 );
        timer_ = this->create_wall_timer(0.5s , std::bind(&PublisherTest::publsish_data , this));
        RCLCPP_INFO(this->get_logger() , "publsiher started");
    }
private:
    void publsish_data(){
        auto msg = example_interfaces::msg::String();
        msg.data = "Hello from cpp pub";
        publisher_->publish(msg);
    }

  rclcpp::Publisher<example_interfaces::msg::String>::SharedPtr publisher_;
  rclcpp::TimerBase::SharedPtr timer_;
};


int main(int argc , char **argv)
{
    rclcpp::init(argc , argv);
    auto node = std::make_shared<PublisherTest>();
    rclcpp::spin(node);
    rclcpp::shutdown();

    return 0;
} 

File: src/my_py_pkg/my_py_pkg/publisher_test.py
import rclpy
from rclpy.node import Node
from example_interfaces.msg import String

class PublisherTest(Node):
    def __init__(self):
        super().__init__("publisher_test")
        self.publisher_ = self.create_publisher(
            String , "my_topic" , 10 
        )
        self.timer_ = self.create_timer(
            0.5 , self.publish_data
        )
        self.get_logger().info("Publisher started!")

    def publish_data(self):
        msg = String()
        msg.data = "Helloooo"
        self.publisher_.publish(msg)


def main(args=None): 
    rclpy.init(args=args)
    node = PublisherTest()
    rclpy.spin(node)
    rclpy.shutdown() 

if __name__ == '__main__':
    main()           